### ISSUE TITLE
Dynamic galleries

### ISSUE THREAD

**Issue by fbmarques** [Ks89/angular-modal-gallery#176]
<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Feature request
[ ] Documentation issue or request
[x] Support request
</code></pre>

<pre><code>
- Node version: 11.5.0
- npm version: 6.4.1
- Operating System and version:  Linux
- Angular version: 7.1.4
- angular-cli version 7.1.4

Browser:
- [x] Chrome (desktop) version XX
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX

Hello!

How do I create galleries dynamically?

When I try to do it, based on the example of "Plain gallery with image pointer.", When I click on the gallery open the amount of galleries you have in * ngFor, all at once.

Grateful for the attention!!

**Comment by Ks89** [comment]
Hi.
Please check the official live example and click on menu's Modal gallery item.
As you can see there is a button to add images at the beginning of the page. This button executes a method called "addRandomImage" in the component's code.

It's very simple, because you only need to add new elements to the array of images, but take care of immutability.
If you try to push images to the array, Angular won't trigger changes and you'll see the same images. If you want to trigger a change when OnPushStrategy is enabled (like in this library) you must create a new instance of the array in a way like this:

```
 addImage() {
    const newImage: Image = .......;
    this.images = [...this.images, newImage];
  }
```

In this snippet I used es6 syntax, but you are free also to use ES5 if you want. The important thing is that you must provide a new instance and not the same array with new element pushed inside.

Feel free to comment if you are still having the same problem

**Comment by Ks89** [comment]
Oh, wait probably I didn't understand your problem. If not, please explain it again.

**Comment by fbmarques** [comment]
Ks89, thank you very much for your attention.

So my problem is this:

I have 3 products, each one needs to have a gallery, I can do it dynamically.

But, what I did not understand is how to call the pictures from each gallery, and dynamically.

In your examples, there are only one images: Image [], I think you could make an example with two galleries, with different images, I think it would help a lot to better understand how things work

Once again thank you for the attention !!!

Atte,
fbmarques

**Comment by Ks89** [comment]
Hi, my official example is a big page with multiple galleries in the same webpage. If you want different images, you should simply create two different arrays. I don't see any problem to do this.
Every instance of modal-gallery can use a different array of images.

**Comment by fbmarques** [comment]
I understand...

Could you tell me the example you mentioned with a sample link?

Once again thank you for the attention !!!

Atte,
fbmarques

**Comment by Ks89** [comment]
This is the live example. As you can see I'm using multiple instances of the gallery, sometime with the same array, sometime not.
https://stackblitz.com/edit/angular-modal-gallery-v7

**Comment by Hirbod** [comment]
@Ks89 I guess he has the same issue as I have: how to create dynamic galleries (multiple) with the PLAIN format. I can't see any method to pass a gallery-id.

I also need 20 Plain-Galleries.

**Comment by Ks89** [comment]
Please check my live example linked above. There is also a documentation website with tons of examples.
Id is mandatory input of the gallery, called exactly 'id'.

If I didn't understand your ploblem yet, please try to explain it again in a better way.

**Comment by fbmarques** [comment]
@Hirbod I also could not do it dynamically.

@Ks89 has examples, but it uses the same gallery, I need to make the call via the database, where it has the reference of the image, in the case, it uses google images, not coming from the database.

The ID is important, but the gallery call is not by the id but by the [modalImages] = "images"

It would be very interesting an example where someone gets images with paths referenced by the database.

Grateful for the attention !!

**Comment by Ks89** [comment]
So do you want to init the gallery with an empty array and add images asynchronously?
If yes you could simply create an empty array and add images to the input (not pushing but re-instantiating the array) dynamically. In my examples there is a demo where images are added using an interval function (probably at the end of the live example in modal section)

**Comment by fbmarques** [comment]
@Ks89 again, thank you for your attention!

As I tried to explain, I have products, registered in the database, with descriptions.

Each product has a maximum of 5 photos.

I can bring the products of the database, I put the id in the "ks-modal-gallery", but in [modalImages] = "images" I have already tried in several ways and could not associate by ID or by the name I created of each collection of products.

I could not associate my case with their examples, and apparently other people did not.

Again grateful for the attention!
